pallets/runtime/tests/src/transaction_ordering_test.rs

```diff
@@ -0,0 +1,1056 @@
+    let charlie_did = register_keyring_account(AccountKeyring::Charlie).unwrap();
+
+    // Setup a mercat asset.
+    let token_name = b"ACME";
+    let ticker = Ticker::try_from(&token_name[..]).unwrap();
+
+    // Create an account for Alice and mint `total_supply` tokens to ACME.
+    let (
+        tx_id,
+        alice_secret_account,
+        alice_account_id,
+        alice_public_account,
+        alice_encrypted_init_balance,
+    ) = create_account_and_mint_token(
+        AccountKeyring::Alice.public(), // owner of ACME.
+        alice_did,
+        total_supply,
+        token_name.to_vec(),
+        0,
+        &mut rng,
+    );
+
+    let alice_creds = AccountCredentials {
+        key: AccountKeyring::Alice,
+        did: alice_did,
+        account_id: alice_account_id,
+        public_account: alice_public_account,
+    };
+
+    // Create an account for Charlie.
+    let tx_id = tx_id + 1;
+    let (charlie_secret_account, _, charlie_public_account, _) =
+        init_account(tx_id, &mut rng, token_name, charlie, charlie_did);
+    let charlie_creds = MediatorCredentials {
+        mediator_key: AccountKeyring::Charlie,
+        mediator_did: charlie_did,
+        mediator_public_account: charlie_public_account,
+        ticker: ticker,
+    };
+
+    (
+        alice_creds,
+        alice_secret_account,
+        alice_encrypted_init_balance,
+        charlie_creds,
+        charlie_secret_account,
+    )
+}
+
+fn create_investor_account(
+    tx_id: u32,
+    key: AccountKeyring,
+) -> (SecAccount, AccountCredentials, EncryptedAmount) {
+    let mut rng = StdRng::from_seed([10u8; 32]);
+    let token_name = b"ACME";
+    // Create accounts for the key holder.
+    let did = register_keyring_account(key).unwrap();
+
+    let (secret_account, account_id, public_account, init_balance) =
+        init_account(tx_id, &mut rng, token_name, key.public(), did);
+
+    let creds = AccountCredentials {
+        key: key,
+        did: did,
+        account_id: account_id,
+        public_account: public_account,
+    };
+
+    (secret_account, creds, init_balance)
+}
+
+#[test]
+fn settle_out_of_order() {
+    ExtBuilder::default()
+        .cdd_providers(vec![AccountKeyring::Eve.public()])
+        .set_max_legs_allowed(500)
+        .build()
+        .execute_with(|| {
+            // Setting:
+            //   - Alice is the token issuer, and has 10 assets in her supply.
+            //   - Bob has a normal account.
+            //   - Charlie is the mediator.
+            //   - Eve is the CDD provider.
+            let (
+                alice_creds,
+                alice_secret_account,
+                alice_init_balance,
+                charlie_creds,
+                charlie_secret_account,
+            ) = chain_set_up(10u128);
+
+            let (bob_secret_account, bob_creds, bob_init_balance) =
+                create_investor_account(2u32, AccountKeyring::Bob);
+
+            let tx_id = 1000u32;
```

Comment:
transaction id should be instruction id (I have made the same mistake in my tests, can you fix those as well?).

Reply:
These `tx_id` are only used by the mercat library. Honestly, I think we should just remove them altogether. There was a time we were signing transactions in the library, and it made sense to sign the `tx_id` to prevent replay attacks, now that's done on the Polymesh side.

Reply:
Yeah makes sense to remove it altogether.